Update readme.md (#1)
SergeyNakhankov/asp-net-core-dashboard-recalculate-totals: readme.md

# Dashboard for ASP.NET Core - How to Recalculate Totals when You Filter a Grid

> The Dashboard does not fully support Grid column filters out of the box. These filters can only be used to select which data to show in the Grid item and they do not affect Total values calculated by the dashboard data engine. As a result, when you apply grid column filters, the Grid item shows filtered data, but Totals remain the same.

Reviewer: ```suggestion
The Dashboard does not fully support Grid column filters out of the box. These filters can only be used to select which data to show in the Grid item and they do not affect Total values calculated by the dashboard data engine. As a result, the Grid item shows filtered data when you apply grid column filters, but Totals are unchanged.
```

The example shows how to recalculate totals when when you apply column filters to the Grid dashboard item.

> > The example supports all the dashboard's [Total types](https://docs.devexpress.com/Dashboard/117302/web-dashboard/create-dashboards-on-the-web/dashboard-item-settings/grid/totals#totals-overview) except for "Auto". If you apply the "Auto" total type in the dashboard, the "Not Supported" text will be shown instead of the total value.

Reviewer: ```suggestion
> The example supports all the dashboard's [Total types](https://docs.devexpress.com/Dashboard/117302/web-dashboard/create-dashboards-on-the-web/dashboard-item-settings/grid/totals#totals-overview) except for "Auto". If you apply the "Auto" total type in the dashboard, "Not Supported" is shown instead of the total value.
```

## Example Overview
### Client

> The [CustomGridTotalsExtension.js](./CS/AspNetCoreDashboard_RecalculateTotals/wwwroot/js/CustomGridTotalsExtension.js) custom extension implements the required functionality on the client. The `dxDataGrid`'s [Total Summary](https://js.devexpress.com/Documentation/Guide/UI_Components/DataGrid/Summaries/Total_Summary/) is used to modify the Grid item's underlying UI component. The [ViewerApiExtensionOptions.onItemWidgetOptionsPrepared](https://docs.devexpress.com/Dashboard/js-DevExpress.Dashboard.ViewerApiExtensionOptions?p=netframework#js_devexpress_dashboard_viewerapiextensionoptions_onitemwidgetoptionsprepared) event allows you to access underlying `dxDataGrid` UI component and change its options.

Reviewer: ```suggestion
The [CustomGridTotalsExtension.js](./CS/AspNetCoreDashboard_RecalculateTotals/wwwroot/js/CustomGridTotalsExtension.js) custom extension implements the required functionality on the client. The `dxDataGrid`'s [Total Summary](https://js.devexpress.com/Documentation/Guide/UI_Components/DataGrid/Summaries/Total_Summary/) is used to modify the Grid item's underlying UI component. The [ViewerApiExtensionOptions.onItemWidgetOptionsPrepared](https://docs.devexpress.com/Dashboard/js-DevExpress.Dashboard.ViewerApiExtensionOptions?p=netframework#js_devexpress_dashboard_viewerapiextensionoptions_onitemwidgetoptionsprepared) event allows you to access the underlying `dxDataGrid` UI component and change its options.
```
